### PR TITLE
Intertrade 8 add user crud operations

### PR DESCRIPTION
**Pull request by Gogi4aaa**
implemented:
1.Get All users functionality
2.Add user functionality
3.Update user and related address functionality
4.Delete user and related address functionality

### CODE REVIEW

File: api/services/user.service.js [hunk 1]
```diff
@@ -36,45 +40,93 @@ export const register = async (req, res, next) => {
...
+      return next(Conflict.invalidUser());
+    }
+    var addresId = user.addressID;
+    if (!addresId) {
+      return next(Conflict.invalidAddress());
+    }
+    await userRepository.deleteUser(userId);
+    await addressService.deleteAddress(addresId);
+  } catch {
+    return next(new UnexpectedError());
```
Comment (alisyedn) [inline api/services/user.service.js:129]: what would throw this?

File: api/services/user.service.js [hunk 2]
```diff
@@ -36,45 +40,93 @@ export const register = async (req, res, next) => {
...
+    //TODO update address information before updating the user!!!
+    updatedUser = await userRepository.updateUser(userData, userId);
+  } catch {
+    return next(new UnexpectedError());
+  }
 
-  return res.status(201)
+  res.status(200).json(updatedUser);
+};
+export const deleteUser = async (req, res, next) => {
```
Comment (alisyedn) [inline api/services/user.service.js:115]: do we need to be able to remove users?

Comment (Gogi4aaa) [reply api/services/user.service.js:115]: yes administrator should have this feature

Comment (Gogi4aaa) [reply api/services/user.service.js:115]: Also if user wants to delete his account permanently from our system

File: api/services/user.service.js (hunk 1 above)
Comment (Gogi4aaa) [reply api/services/user.service.js:129]: this throws server status 500

File: api/repositories/user.repository.js [hunk 3]
```diff
@@ -1,23 +1,46 @@
...
     where: {email},
     select: {
       id: true
     }
   })
+  return user?.id;
+}
+//user crud Operations
+export const getAllUsers = async () => {
+  const users = await prisma.users.findMany();
```
Comment (alisyedn) [inline api/repositories/user.repository.js]: u can just return here

File: api/errors/Conflict.js [hunk 4]
```diff
@@ -3,16 +3,22 @@ import {ApplicationError} from "./ApplicationError.js";
...
   static usernameAlreadyExists() {
-    return new Conflict("This username already exists!", Conflict.TYPE_USERNAME_ALREADY_EXISTS);
+    return new Conflict("This username already exists!", this.TYPE_USERNAME_ALREADY_EXISTS);
   }
-
   static emailAlreadyExists() {
-    return new Conflict("This email already exists!", Conflict.TYPE_EMAIL_ALREADY_EXISTS);
+    return new Conflict("This email already exists!", this.TYPE_EMAIL_ALREADY_EXISTS);
+  }
+  static invalidUser() {
```
Comment (alisyedn) [inline api/errors/Conflict.js:18]: y would a invalid user be conflict?

File: api/errors/Conflict.js [hunk 5]
```diff
@@ -3,16 +3,22 @@ import {ApplicationError} from "./ApplicationError.js";
...
+    return new Conflict("Invalid user!", this.TYPE_USER_INVALID);
+  }
+  static invalidAddress() {
```
Comment (alisyedn) [inline api/errors/Conflict.js:21]: same here is Conflict the best status for this?

File: api/errors/Conflict.js (hunk 4 above)
Comment (Gogi4aaa) [reply api/errors/Conflict.js:18]: what would it be

File: api/errors/Conflict.js (hunk 5 above)
Comment (Gogi4aaa) [reply api/errors/Conflict.js:21]: is 401 best here?

**Comment by cruckman900** [comment]
looking good! ready to go from me

**Pull request merged**